ibis/backends/tests/test_generic.py

```diff
@@ -760,9 +769,406 @@ def test_exists(batting, awards_players, method_name):
+                    ['duckdb'],
+                    "<class 'ibis.expr.datatypes.core.INET'>",
+                    raises=KeyError,
+                ),
+            ],
+            id="inet",
+        ),
+        param(
+            ibis.literal(decimal.Decimal("1.2"), type=dt.decimal),
+            {
+                'bigquery': "INT64",
+                'snowflake': "VARCHAR",
+                "duckdb": "DECIMAL(18,3)",
+                'sqlite': "real",
+                "postgres": "numeric",
+            },
+            id="decimal",
+            marks=[
+                pytest.mark.broken(
+                    ['bigquery'],
+                    "ufunc 'isfinite' not supported for the input types, and the inputs could not be "
+                    "safely coerced to any supported types according to the casting rule ''safe''",
+                    raises=TypeError,
+                ),
+            ],
+        ),
+        param(
+            ibis.array([1.0, 2.0, 3.0]),
+            {
+                "bigquery": "ARRAY",
+                'snowflake': "ARRAY",
+                "duckdb": "DOUBLE[]",
+                "postgres": "numeric[]",
+            },
+            marks=[
+                pytest.mark.broken(
+                    ['sqlite'],
+                    "(sqlite3.InterfaceError) Error binding parameter 0 - probably unsupported type.",
+                    raises=sqlalchemy.exc.InterfaceError,
+                ),
+            ],
+            id="array<float>",
+        ),
+        param(
+            ibis.array(["A", "B", "C"]),
+            {
+                "bigquery": "ARRAY",
+                'snowflake': "ARRAY",
+                "duckdb": "VARCHAR[]",
+                "postgres": "text[]",
+            },
+            marks=[
+                pytest.mark.broken(
+                    ['sqlite'],
+                    "(sqlite3.InterfaceError) Error binding parameter 0 - probably unsupported type.",
+                    raises=sqlalchemy.exc.InterfaceError,
+                ),
+            ],
+            id="array<string>",
+        ),
+        # TODO: We should add test cases for other types including:
+        #   dt.geometry, dt.geography, dt.point, dt.linestring, dt.polygon, dt.multilinestring,
+        #   dt.multipoint, dt.multipolygon, dt.json, dt.category, dt.struct
+    ],
+)
+@pytest.mark.notimpl(
+    [
+        "dask",
+        "datafusion",
+        "mssql",
+        "mysql",
+        "pandas",
+        "polars",
+        "pyspark",
+        "trino",
+    ],
+    raises=(NotImplementedError, com.OperationNotDefinedError),
+    strict=True,
+)
+def test_typeof(backend, con, ibis_op, expected_values):
+    expr = ibis_op.typeof()
+    result = con.execute(expr)
+
+    backend_name = backend.name()
+    if backend_name in expected_values:
+        assert result == expected_values[backend_name]
+    else:
+        raise Exception(f"Value for backend is undefined. Current value: {result!r}")
```

Comment:
Can this be `xfail`d or `skip`ped instead?

Reply:
No. This is not a backend bug but a bug in the test and we should just add the missing value to the test. 

Before, I had code like below:
```
assert result == expected_values[backend_name]
```
But that didn't display the expected value, and it threw a KeyError exception. Now that I have the expected value, I can update the tests based on the CI results and I don't have to configure all environments.

Before merging, we can bring back the original form.